tests/components/rfxtrx/test_light.py

```diff
@@ -147,82 +156,70 @@ async def test_several_lights(hass, rfxtrx):
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Living"
+
+    assert len(hass.states.async_all()) == 3
+
+
+@pytest.mark.parametrize("repetitions", [1, 3])
+async def test_repetitions(hass, rfxtrx, repetitions):
+    """Test signal repetitions."""
     await async_setup_component(
         hass,
         "light",
-        {"light": {"platform": "rfxtrx", "automatic_add": True, "devices": {}}},
+        {
+            "light": {
+                "platform": "rfxtrx",
+                "signal_repetitions": repetitions,
+                "devices": {"0b1100cd0213c7f230010f71": {"name": "Test"}},
+            }
+        },
     )
     await hass.async_block_till_done()
 
-    event = rfxtrx_core.get_rfx_object("0b11009e00e6116202020070")
-    event.data = bytearray(b"\x0b\x11\x00\x9e\x00\xe6\x11b\x02\x02\x00p")
-
-    await _signal_event(hass, event)
-    entity = rfxtrx_core.RFX_DEVICES["0e61162_2"]
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
-    assert "<Entity 0b11009e00e6116202020070: on>" == entity.__str__()
-
-    event = rfxtrx_core.get_rfx_object("0b11009e00e6116201010070")
-    event.data = bytearray(b"\x0b\x11\x00\x9e\x00\xe6\x11b\x01\x01\x00p")
+    await hass.services.async_call(
+        "light", "turn_on", {"entity_id": "light.test"}, blocking=True
+    )
+    await hass.async_block_till_done()
 
-    await _signal_event(hass, event)
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
+    assert rfxtrx.transport.send.call_count == repetitions
 
-    event = rfxtrx_core.get_rfx_object("0b1100120118cdea02020070")
-    event.data = bytearray(
-        [0x0B, 0x11, 0x00, 0x12, 0x01, 0x18, 0xCD, 0xEA, 0x02, 0x02, 0x00, 0x70]
-    )
 
-    await _signal_event(hass, event)
-    entity = rfxtrx_core.RFX_DEVICES["118cdea_2"]
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
-    assert "<Entity 0b1100120118cdea02020070: on>" == entity.__str__()
-
-    # trying to add a sensor
-    event = rfxtrx_core.get_rfx_object("0a52085e070100b31b0279")
-    event.data = bytearray(b"\nR\x08^\x07\x01\x00\xb3\x1b\x02y")
-    await _signal_event(hass, event)
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
-
-    # trying to add a switch
-    event = rfxtrx_core.get_rfx_object("0b1100100118cdea02010f70")
-    event.data = bytearray(
-        [0x0B, 0x11, 0x00, 0x10, 0x01, 0x18, 0xCD, 0xEA, 0x01, 0x01, 0x0F, 0x70]
+async def test_discover_light(hass, rfxtrx):
+    """Test with discovery of lights."""
+    await async_setup_component(
+        hass,
+        "light",
+        {"light": {"platform": "rfxtrx", "automatic_add": True, "devices": {}}},
     )
+    await hass.async_block_till_done()
 
-    await _signal_event(hass, event)
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
+    await _signal_event(hass, "0b11009e00e6116202020070")
+    state = hass.states.get("light.0b11009e00e6116202020070")
+    assert state
+    assert state.state == "on"
+    assert state.attributes.get("friendly_name") == "0b11009e00e6116202020070"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)